src/test/scala/chiselTests/aop/SelectSpec.scala

```diff
@@ -228,4 +229,68 @@ class SelectSpec extends ChiselFlatSpec {
     intercept[Exception] { Select.instances(top) }
   }
 
+  "Select currentInstancesIn and allCurrentInstancesIn" should "support module, extmodule, and D/I" in {
+    import chisel3.experimental.hierarchy._
+
+    class MyLeafExtModule extends ExtModule {}
+
+    class MyLeafModule extends RawModule {}
+
+    class MyIntermediateModule extends RawModule {
+      Module(new MyLeafModule)
+      Module(new MyLeafModule)
+    }
+
+    @instantiable
+    class MyLeafInstance extends RawModule {}
+
+    @instantiable
+    class MyIntermediateInstance extends RawModule {
+      val definition = Definition(new MyLeafInstance)
+      Instance(definition)
+      Instance(definition)
+    }
+
+    class MyWrapperModule extends RawModule {
+      val definition = Definition(new MyIntermediateInstance)
+      Module(new MyIntermediateModule)
+      Instance(definition)
+      Module(new MyLeafExtModule)
+
+      atModuleBodyEnd {
```

Comment:
You don't really need `atModuleBodyEnd` since you've put this after the children modules.

Also, I'd like to see a test that runs before all Modules have been constructed, so maybe add another check between the `Instance` and one of the Modules?

Reply:
Yep sounds good.